sensor_board/common/pinav_parser.c

```diff
@@ -0,0 +1,266 @@
+#include "pinav_parser.h"
+#include <string.h>
+#include <stdio.h>	// @TODO: remove when not needed for debugging
+
+typedef enum generic_status { FAIL = 0, SUCCESS = 1 } generic_status_t;
+
+// Private helper function to perform initial input validity checks
+static pinav_parser_status_t verify_inputs(pinav_parse_output_t * out, uint8_t * sentence) {
+	// Verify pointers
+	if (!out) {
+		return NULL_OUTPUT_PTR;
+	}
+	if (!sentence) {
+		out->id = NONE;
+		return NULL_SENTENCE_PTR;
+	}
+
+	// Verify that the sentence starts with '$'
+	if (*sentence != '$') {
+		out->id = NONE;
+		return SENTENCE_FORMAT_ERROR;
+	}
+
+	// Verify that the CR+LF sequence is within the 
+	// 80 charater limit specified by the datasheet
+	// @TODO Verify whether the CR+LF is included in the 80 char total
+	uint8_t * it;
+	uint8_t found = 0;
+	for (it = sentence; (it < sentence + 79) && (*it != '\0'); ++it) {
+		if ((*it == '\r') && (*(it + 1) == '\n')) {
+			found = 1;
+			break;
+		}
+	}
+	if (!found) {
+		out->id = NONE;
+		return IMPROPER_SENTENCE_LENGTH;
+	}
+
+	return OK;
+}
+
+// Private helper function to determine whether a string can be converted to a number
+static generic_status_t can_convert_stoi(const uint8_t * s, size_t len) {
```

Comment:
I'm not sure how I feel about this function. It seems like their functionality can be rolled in to `stoi`, which would reduce the cost of performing these conversions and the amount of code we need to maintain without losing any modularity. Perhaps `can_convert_soti` could be provided separately if we find that we need it somewhere else, but as things stand it seems like its only used in the conversion function.